### PR TITLE
GOLD-238 fix: sql injection when recordTxStatus feature is on

### PR DESCRIPTION
**Pull request by kgmyatthu**

### CODE REVIEW

File: src/middlewares/injectIP.ts [hunk 1]
```diff
@@ -2,7 +2,14 @@
 import { NextFunction, Request, Response } from 'express'
 
 const injectIP = (req: Request, res: Response, next: NextFunction): void => {
-  if (req.body.method === 'eth_sendRawTransaction' && CONFIG.recordTxStatus) req.body.params[1000] = req.ip
+  if (req.body.method === 'eth_sendRawTransaction' && CONFIG.recordTxStatus){
+    // prevents sql injection
+    const regex_str = "^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}$"
```
[the remark this answers is not in the document]
Comment (kgmyatthu) [reply src/middlewares/injectIP.ts]: Addressed in new commit

[the remark this answers is not in the document]
Comment (kgmyatthu) [reply src/middlewares/injectIP.ts]: Addressed in new commit

[the remark this answers is not in the document]
Comment (kgmyatthu) [reply src/middlewares/injectIP.ts]: Addresssed in new commit

[the remark this answers is not in the document]
Comment (kgmyatthu) [reply src/middlewares/injectIP.ts]: Addressed

**Review by arhamj — APPROVED** [review]

**Review by afostr — APPROVED** [review]
looks great, thanks

**Pull request merged**